core/src/main/java/com/amazon/opendistroforelasticsearch/sql/expression/operator/arthmetic/MathematicalFunction.java

```diff
@@ -0,0 +1,214 @@
+import com.amazon.opendistroforelasticsearch.sql.expression.function.BuiltinFunctionRepository;
+import com.amazon.opendistroforelasticsearch.sql.expression.function.FunctionBuilder;
+import com.amazon.opendistroforelasticsearch.sql.expression.function.FunctionName;
+import com.amazon.opendistroforelasticsearch.sql.expression.function.FunctionResolver;
+import com.amazon.opendistroforelasticsearch.sql.expression.function.FunctionSignature;
+import com.google.common.collect.ImmutableMap;
+import java.util.Arrays;
+import java.util.Map;
+import java.util.function.BiFunction;
+import java.util.function.Function;
+import lombok.experimental.UtilityClass;
+
+@UtilityClass
+public class MathematicalFunction {
+  /**
+   * Register Mathematical Functions.
+   *
+   * @param repository {@link BuiltinFunctionRepository}.
+   */
+  public static void register(BuiltinFunctionRepository repository) {
+    repository.register(abs());
+    repository.register(ceil());
+    repository.register(ceiling());
+    repository.register(exp());
+    repository.register(floor());
+    repository.register(ln());
+    repository.register(log());
+  }
+
+  /**
+   * Definition of abs() function.
+   * The supported signature of abs() function are
+   * INT -> INT
+   * LONG -> LONG
+   * FLOAT -> FLOAT
+   * DOUBLE -> DOUBLE
+   */
+  private static FunctionResolver abs() {
+    return new FunctionResolver(
+        BuiltinFunctionName.ABS.getName(),
+        singleArgumentFunction(
+            BuiltinFunctionName.ABS.getName(), Math::abs, Math::abs, Math::abs, Math::abs));
+  }
+
+  /**
+   * Definition of ceil(x)/ceiling(x) function.
+   * Calculate the next highest integer that x rounds up to
+   * The supported signature of ceil/ceiling function is
+   * DOUBLE -> LONG
+   */
+  private static FunctionResolver ceil() {
+    FunctionName functionName = BuiltinFunctionName.CEIL.getName();
+    return new FunctionResolver(
+        BuiltinFunctionName.CEIL.getName(),
+        new ImmutableMap.Builder<FunctionSignature, FunctionBuilder>()
+            .put(
+                new FunctionSignature(functionName, Arrays.asList(ExprType.DOUBLE)),
+                unaryOperator(
+                    functionName, v -> ((long) Math.ceil(v)), ExprValueUtils::getDoubleValue,
+                    ExprType.LONG))
+            .build());
+  }
+
+  private static FunctionResolver ceiling() {
+    FunctionName functionName = BuiltinFunctionName.CEILING.getName();
+    return new FunctionResolver(
+        BuiltinFunctionName.CEILING.getName(),
+        new ImmutableMap.Builder<FunctionSignature, FunctionBuilder>()
+            .put(
+                new FunctionSignature(functionName, Arrays.asList(ExprType.DOUBLE)),
+                unaryOperator(
+                    functionName, v -> ((long) Math.ceil(v)), ExprValueUtils::getDoubleValue,
+                    ExprType.LONG))
+            .build());
```

Comment:
Exact same as `ceil()`?

Reply:
Yes the same method but with different registered name

Reply:
I see. I think we can refactor these kind of similar function family later by generating from same method with different argument, such as substr and substring, log and log2 etc.